src/hotspot/share/utilities/globalDefinitions_gcc.hpp

```diff
@@ -215,6 +215,12 @@ inline int g_isnan(double f) { return isnan(f); }
 #define CAN_USE_NAN_DEFINE 1
 #endif
 
+#ifdef __APPLE__
+#include <TargetConditionals.h>
```

Comment:
There are a lot of instances of this pattern.  Is there a platform specific header you can add that can be used instead?

Reply:
If it considered less intrusive, I can do that.
But it spans different components (e.g. hotspot, libjava, libjli,...)

Reply:
Hopefully there are only 2 header files (one for hotspot and one for jdk sources) that you'd need to add this.  globalDefinitions_gcc.hpp is the right place for hotspot, not sure about the jdk.

Reply:
A better solution might be to add the define in open/make/autoconf/flags-cflags.m4 in FLAGS_SETUP_CFLAGS_HELPER.  This would reduce the changes to a single file.

Reply:
True, that sounds much better.
I updated the PR.